File: src/Services/API/Domain/Table/TableMetadata.cs
namespace API.Domain.Table
{
    public class TableMetadata
    {
        public IEnumerable<TableColumns> Columns { get; set; }
        public TableCreateScript CreateScript { get; set; }
        public IEnumerable<TableDescription> Descriptions { get; set; }
        public IEnumerable<TableIndex> Indices { get; set; }
        public IEnumerable<TableForeignKey> ForeignKeys { get; set; }
        public IEnumerable<TableProperty> Properties { get; set; }
        public IEnumerable<TableConstraint> Constraints { get; set; }
        public IEnumerable<TableFragmentation> TableFragmentations { get; set; }
        public string TableDependencies { get; set; }
    }
}

File: src/WebClient/mssql.client/src/app/table/models/TableMetadata.ts
import { TableConstraint } from './TableConstraint';
import { TableIndex } from './TableIndex';
import { TableForeignKey } from './TableForeignKey';
import { TableProperty } from './TableProperty';
import { TableDescription } from './TableDescription';
import { TableColumn } from './TableColumn';
import { TableCreateScript } from './TableCreateScript';
import { TableFragmentation } from './TableFragmentation';

export interface TableMetadata {
  columns: TableColumn[];
  createScript: TableCreateScript;
  descriptions: TableDescription[];
  indices: TableIndex[];
  foreignKeys: TableForeignKey[];
  properties: TableProperty[];
  constraint: TableConstraint[];
  tableFragmentations: TableFragmentation[];
  tableDependenciesTree: string;
}
